### PR TITLE
Update K8s Security CS with Service mesh & OPA

### PR DESCRIPTION
**Pull request by ragashreeshekar**
Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [ ] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [ ] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [ ] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [ ] All your assets are stored in the **assets** folder.
- [ ] All the images used are in the **PNG** format.
- [ ] Any references to websites have been formatted as [TEXT](URL)
- [ ] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [ ] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md [hunk 1]
```diff
@@ -270,6 +270,54 @@ spec:
 
 For more information on security context for Pods, refer to the documentation at https://kubernetes.io/docs/tasks/configure-pod-container/security-context/
 
+### Implement Service Mesh
```
Comment (mackowski) [inline cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md:273]: can you describe some disadvantages of using service mesh?

File: cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md [hunk 2]
```diff
@@ -270,6 +270,54 @@ spec:
...
+
+#### Operational Control
+A service mesh allows security and platform teams to set the right macro controls to enforce access controls, while allowing developers to make customizations they need to move quickly within these guardrails.
+
+#### RBAC
+A strong Role Based Access Control (RBAC) system is arguably one of the most critical requirements in large engineering organizations, since even the most secure system can be easily circumvented by overprivileged users or employees. Restricting privileged users to least privileges necessary to perform job responsibilities, ensuring access to systems are set to “deny all” by default, and ensuring proper documentation detailing roles and responsibilities are in place is one of the most critical security concerns in the enterprise.
+
+### Implementing Open Policy Agent (OPA) for a centralized policy management
```
Comment (mackowski) [inline cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md:296]: any alternatives for it?

**Review by mackowski — CHANGES_REQUESTED** [review]
Looks very good thank you.

**Comment by Seppl2202** [comment]
> Ingress & Egress Control Service mesh adds a layer of security that allows you to monitor and address compromising traffic as it enters the mesh. Istio integrates with Kubernetes as an ingress controller and takes care of load balancing for ingress. This allows you to add a level of security at the perimeter with ingress rules. Egress control allows you to see and manage external services and control how your services interact with them.

While writing my bachelor thesis, I encountered massive issues when using Istio together with Ranger and Azure automization of certificates of exposed services (e.g. a web app): https://github.com/istio/istio/issues/7776

The reason was that Istio used a ressource called "Gateway" to expose a service, while the Azure automatization required an "Ingress" ressource to recognize it for certificate creation.
I read that Istio now has an "Ingress" ressource (https://istio.io/latest/docs/tasks/traffic-management/ingress/), however I am not really familiar with that one. Maybe someone has further information about that?

> Pod Security Policies are one way to control the security-related attributes of pods, including container privilege levels. These can allow an operator to specify the following:
> 
> Do not run application processes as root.
> Do not allow privilege escalation.
> Use a read-only root filesystem.
> Use the default (masked) /proc filesystem mount
> **Do not use the host network or process space.**

--> I would clearly state that using "hostNetwork:true" will cause NetworkPolicies to be ignored since the Pod will use its host (=the node it runs on) network (The Git issue is from Calico but I encountered that one with all CNI providers I used): https://github.com/projectcalico/felix/issues/1361.
This can cause serious trouble :)

**Comment by mackowski** [comment]
@ragashreeshekar please address @Seppl2202  and my comments. If you need any help let us know.

File: cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md [hunk 3]
```diff
@@ -611,4 +658,7 @@ Master documentation - https://kubernetes.io/
 10. 12 Kubernetes configuration best practices - https://www.stackrox.com/post/2019/09/12-kubernetes-configuration-best-practices/#6-securely-configure-the-kubernetes-api-server
 11. A Practical Guide to Kubernetes Logging - https://logz.io/blog/a-practical-guide-to-kubernetes-logging/
 12. Kubernetes Web UI (Dashboard) - https://kubernetes.io/docs/tasks/access-application-cluster/web-ui-dashboard/	
-13. Tesla cloud resources are hacked to run cryptocurrency-mining malware - https://arstechnica.com/information-technology/2018/02/tesla-cloud-resources-are-hacked-to-run-cryptocurrency-mining-malware/
\ No newline at end of file
+13. Tesla cloud resources are hacked to run cryptocurrency-mining malware - https://arstechnica.com/information-technology/2018/02/tesla-cloud-resources-are-hacked-to-run-cryptocurrency-mining-malware/
```
[the remark this answers is not in the document]
Comment (ragashreeshekar) [reply cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md:669]: Bits and parts of the listed resources are utilized across the entirety of the draft to build the story line hence want to confirm if you'd want me to work on referencing it throughout? Please advise

File: cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md (hunk 1 above)
Comment (ragashreeshekar) [reply cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md:274]: Done, in commit https://github.com/OWASP/CheatSheetSeries/pull/496/commits/a1275b258856778c2e73a641b780e359b1a1b4b7

File: cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md (hunk 2 above)
Comment (ragashreeshekar) [reply cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md:304]: This topic was one of suggestions and I'm not quite exposed to it. I did some research however couldn't find much resource. Could the team pls help with it? Thanks in advance

**Comment by ragashreeshekar** [comment]
I have updated the 

> > Ingress & Egress Control Service mesh adds a layer of security that allows you to monitor and address compromising traffic as it enters the mesh. Istio integrates with Kubernetes as an ingress controller and takes care of load balancing for ingress. This allows you to add a level of security at the perimeter with ingress rules. Egress control allows you to see and manage external services and control how your services interact with them.
> 
> While writing my bachelor thesis, I encountered massive issues when using Istio together with Ranger and Azure automization of certificates of exposed services (e.g. a web app): [istio/istio#7776](https://github.com/istio/istio/issues/7776)
> 
> The reason was that Istio used a ressource called "Gateway" to expose a service, while the Azure automatization required an "Ingress" ressource to recognize it for certificate creation.
> I read that Istio now has an "Ingress" ressource (https://istio.io/latest/docs/tasks/traffic-management/ingress/), however I am not really familiar with that one. Maybe someone has further information about that?
> 
> > Pod Security Policies are one way to control the security-related attributes of pods, including container privilege levels. These can allow an operator to specify the following:
> > Do not run application processes as root.
> > Do not allow privilege escalation.
> > Use a read-only root filesystem.
> > Use the default (masked) /proc filesystem mount
> > **Do not use the host network or process space.**
> 
> --> I would clearly state that using "hostNetwork:true" will cause NetworkPolicies to be ignored since the Pod will use its host (=the node it runs on) network (The Git issue is from Calico but I encountered that one with all CNI providers I used): [projectcalico/felix#1361](https://github.com/projectcalico/felix/issues/1361).
> This can cause serious trouble :)

Thanks for your review and suggestions. Please advise how we could handle the "Ingress & Egress Control using service mesh", as far as I understand from the review, It may be better to remove that section. Kindly let me know. 

I have updated the draft with the second comment, updated version commit - https://github.com/OWASP/CheatSheetSeries/pull/496/commits/1584fddd03f44487ced5c4e1022255b93dc4af81 

Thanks again!

**Comment by ragashreeshekar** [comment]
Thanks @Seppl2202 @mackowski for your time and review, part of the review comments have been addressed, I have added comments where I need some clarifications, please go through them and let me know your thoughts.

**Comment by mackowski** [comment]
Thanks @ragashreeshekar!

File: cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md (hunk 2 above)
Comment (mackowski) [reply cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md:304]: No worries if you done fair research and you did not find anything that is fine. OPA is kind of the standard now so I am happy that you covered it here.

File: cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mackowski) [reply cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md:669]: Ok, that is fine ;-) we can leave it as is.

**Comment by ragashreeshekar** [comment]
If it looks good, could we merge it to the active CS? 
Please let us know how we could proceed if otherwise. Thanks in advance.

File: cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md (hunk 2 above)
Comment (ragashreeshekar) [reply cheatsheets_draft/Kubernetes_Security_Cheat_Sheet.md:304]: Thanks @mackowski.
 I have done the research to the best of my understanding, would be happy to add details based of the team's guidance.

**Comment by mackowski** [comment]
Yeh it looks well. I will merge it. Thank you @ragashreeshekar for your woirk!

**Review by mackowski — APPROVED** [review]
LGTM

**Pull request merged**